Tests/Pagination/Decorator/LimitStopScrollerDecoratorTest.php

```diff
@@ -96,4 +96,34 @@ public function testInvalid2()
     {
         new LimitStopScrollerDecorator(new NoScroller(), ['limitStop' => ['count' => 12, 'field' => 'whatever']]);
     }
+
+    public function testCloneScrollerDecorator()
+    {
+        $client = new RestClient(new NullLogger());
+        $jobConfig = new JobConfig(['endpoint' => 'test']);
+
+        $config = ['limitStop' => ['count' => 12]];
+
+        $scroller = new PageScroller(['pageParam' => 'pageNo']);
+        $decorator = new LimitStopScrollerDecorator($scroller, $config);
+        $response = new \stdClass();
+        $response->results = (object)['totalNumber' => 15, 'pageNumber' => 1];
+        $response->results->data = array_fill(0, 10, (object)['key' => 'value']);
+
+        $decorator->getNextRequest(
+            $client,
+            $jobConfig,
+            $response,
+            $response->results->data
+        );
+
+        $cloneDecorator = clone $decorator;
+        $cloneDecorator->reset();
+
+        $decoratorState = $decorator->getScroller()->getState();
+        $cloneDecoratorState = $cloneDecorator->getScroller()->getState();
+
+        self::assertEquals(2, $decoratorState['page']);
+        self::assertEquals(1, $cloneDecoratorState['page']);
```

Comment:
btw, ten test prochazi i bez toho clonu :)

Reply:
teď už ne

Reply:
jj, furt

Reply:
to musis zavolat getNextRequest, aby se projevilo, ze je to naklonovany

Reply:
ok, tak jak to myslíš?

Reply:
tak musím zavolat getNextRequest abych se dostal na další stránku a potom když ten objekt naklonuju tak se to projeví že je jeden scroller ve výchozím stavu a druhý je ten původní

Reply:
kdyz vyhodis tohle https://github.com/keboola/juicer/pull/27/files#diff-3e5be99a23bc917cab81e6ed04ea0c6cR25 z kodu, tak tenhle test bude porad zelenej, protoze nestestuje to, ze po getNextRequest se rozejede stav puvodniho a novyho objektu, coz je cil

nebo obracene - nezacal jsi failing testem :)

Reply:
a když ten `__clone()` zakomentuju tak se tím testy rozbijou

Reply:
je, presne tak :) napsali jsme ten komentar zaraz :)

Reply:
aha, me to prochazelo, a ted to neprochazi, chjo, asi cache nebo co, jo, mas pravdu, sorry!

Reply:
hele, tak asi dobry no

Reply:
super :)